.buildkite/scripts/run_job.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python3
+# -*- encoding: utf-8
+
+import argparse
+import os
+import subprocess
+import sys
+
+import git_utils as git
+
+from sbt_dependency_tree import Repository
+
+
+def check_call(cmd):
+    """
+    A wrapped version of subprocess.check_call() that doesn't print a
+    traceback if the command errors.
+    """
+    print("*** Running %r" % " ".join(cmd))
+    try:
+        return subprocess.check_call(cmd)
+    except subprocess.CalledProcessError as err:
+        print(err)
+        sys.exit(err.returncode)
+
+
+def make(*args):
+    """Run a Make command, and check it completes successfully."""
+    check_call(["make"] + list(args))
+
+
+def should_run_sbt_project(repo, project_name, changed_paths):
+    project = repo.get_project(project_name)
+
+    interesting_paths = [p for p in changed_paths if not p.startswith(".sbt_metadata")]
+
+    if ".buildkite/pipeline.yml" in interesting_paths:
+        print("*** Relevant: .buildkite/pipeline.yml")
+        return True
+    if "build.sbt" in interesting_paths:
+        print("*** Relevant: build.sbt")
+        return True
+    if any(p.startswith("project/") for p in interesting_paths):
+        print("*** Relevant: project/")
+        return True
+
+    for path in interesting_paths:
+        if path.startswith((".buildkite", "docs/")):
+            continue
+
+        if path.endswith(".tf"):
```

Comment:
minor: I've just discovered we trigger a full storage-service rebuild for Python scripts which cannot affect SBT projects. 🤦‍♀️ 

```suggestion
        if path.endswith((".py", ".tf")):
```